controllers/util/solr_util.go

```diff
@@ -215,16 +216,63 @@ func GenerateStatefulSet(solrCloud *solr.SolrCloud, solrCloudStatus *solr.SolrCl
 	}
 	// Add backup volumes
 	if solrCloud.Spec.StorageOptions.BackupRestoreOptions != nil {
-		solrVolumes = append(solrVolumes, corev1.Volume{
-			Name:         BackupRestoreVolume,
-			VolumeSource: solrCloud.Spec.StorageOptions.BackupRestoreOptions.Volume,
-		})
-		volumeMounts = append(volumeMounts, corev1.VolumeMount{
-			Name:      BackupRestoreVolume,
-			MountPath: BaseBackupRestorePath,
-			SubPath:   BackupRestoreSubPathForCloud(solrCloud.Spec.StorageOptions.BackupRestoreOptions.Directory, solrCloud.Name),
-			ReadOnly:  false,
-		})
+		backupOptions := solrCloud.Spec.StorageOptions.BackupRestoreOptions
+		// Backup volumes can come from three sources: a singleton volume for local backups (legacy), a list of "managed"
+		// (i.e. local) repositories, and a list of gcs repositories.  All three of these require a volume and
+		// accompanying mount: either for credentials to reach GCS or for the data itself.
+
+		// Handle the "singleton" local volume first
+		if backupOptions.Volume != nil {
+			solrVolumes = append(solrVolumes, corev1.Volume{
+				Name:         solr.BackupRestoreVolume,
+				VolumeSource: *backupOptions.Volume,
+			})
+			volumeMounts = append(volumeMounts, corev1.VolumeMount{
+				Name:      solr.BackupRestoreVolume,
+				MountPath: solr.BaseBackupRestorePath,
+				SubPath:   BackupRestoreSubPathForCloud(backupOptions.Directory, solrCloud.Name),
+				ReadOnly:  false,
+			})
+		}
+
+		// Then handle any other managed volumes
+		if backupOptions.ManagedRepositories != nil {
+			for _, managedRepository := range *backupOptions.ManagedRepositories {
+				solrVolumes = append(solrVolumes, corev1.Volume{
+					Name:         managedRepository.GetVolumeName(),
+					VolumeSource: *managedRepository.Volume,
+				})
+				volumeMounts = append(volumeMounts, corev1.VolumeMount{
+					Name:      managedRepository.GetVolumeName(),
+					MountPath: managedRepository.GetSolrMountPath(),
+					SubPath:   BackupRestoreSubPathForCloud(managedRepository.Directory, solrCloud.Name),
+					ReadOnly:  false,
+				})
+			}
+		}
+
+		// Lastly, handle the volumes needed for any GCS credentials.
+		if backupOptions.GcsRepositories != nil {
+			for _, gcsRepository := range *backupOptions.GcsRepositories {
+				fals := false
+				solrVolumes = append(solrVolumes, corev1.Volume{
+					Name: gcsRepository.GetVolumeName(),
+					VolumeSource: corev1.VolumeSource{
```

Comment:
We might need to worry about restarting the Solr Pods when these secrets update. But again we can have a separate issue/PR for that.